### PR TITLE
Add moto to do s3 unit tests

### PR DESCRIPTION
**Pull request by sesterki**
#### What are the relevant tickets?
None

#### What's this PR do?
Adds package moto and uses it to spoof S3 for unit tests

#### How should this be manually tested?
Rebuild to install moto and then run tests

### CODE REVIEW

File: course_catalog/tasks_test.py [hunk 1]
```diff
@@ -263,8 +270,56 @@ def test_get_mitx_data_unexpected_error(settings):
...
+        with open(TEST_JSON_PATH + "/" + file, 'r') as f:
+            test_bucket.put_object(Key=file_key,
+                                   Body=f.read())
+
+
 def cause_error():
     """
     Helper function created to insert exceptions into otherwise working code
     """
+    print("HERE")
```
Comment (mbertrand) [inline course_catalog/tasks_test.py]: Delete this line

File: requirements.in [hunk 2]
```diff
@@ -16,3 +16,4 @@ uwsgi
 djangorestframework
 django-extensions
 ocw-data-parser==0.1.6
+moto
```
Comment (mbertrand) [inline requirements.in]: Put this in `test_requirements.txt` instead

File: requirements.txt [hunk 3]
```diff
@@ -5,14 +5,18 @@
 #    pip-compile --output-file requirements.txt requirements.in
 #
 amqp==2.3.2               # via kombu
-appnope==0.1.0            # via ipython
+asn1crypto==0.24.0        # via cryptography
```
Comment (mbertrand) [inline requirements.txt]: Revert to previous version without moto

File: course_catalog/tasks_test.py (hunk 1 above)
Comment (sesterki) [reply course_catalog/tasks_test.py]: Deleted

File: requirements.in (hunk 2 above)
Comment (sesterki) [reply requirements.in]: Moved

File: requirements.txt (hunk 3 above)
Comment (sesterki) [reply requirements.txt]: Fixed

**Pull request merged**